.github/workflows/test.yml

```diff
@@ -0,0 +1,40 @@
+name: Run Unit Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  jest:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v4
+
+      - name: Install Node.js, NPM and Yarn
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - name: Install Dependencies
+        run: npm ci
+
+      - name: Install xvfb to provide a virtual display
+        run: sudo apt-get install xvfb
```

Comment:
**Actionable**
This seems to be failing on my local. I'm using a tool called [nektosact/act](https://nektosact.com/usage/index.html).I've attached the error log below. 
[ci_error.log](https://github.com/user-attachments/files/16102170/ci_error.log)

Reply:
Test

Reply:
1. Im trying to use xvfb to run jest headlessly (without display output in a browser). The jest-electron-runner is supposed to be configured to [run a headless instance of electron](https://www.npmjs.com/package/@kayahr/jest-electron-runner#debugging), but for some reason it wasn't working so I opted to just use xvfb.  
The problem might have something to do with the fact that Im using the sudo command to install xvfb - error log says it's having issues resolving the host name to an IP (maybe because nektosact runs actions in a docker container?? not sure). I'll do some tests and get back to you.

2. The second issue with prettier might be because Im using the context property to access the environment variable instead of their [default names](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables). Again, will test this and see if I can find a solution.

BTW nektosact seems like an awesome tool - going to be using this moving forward @oestrada1001

Reply:
Disregard my first comment, most of my assumptions were incorrect lol.

The error with prettier has to do with the fact that the secrets that are usually passed in a Github environement are [not being passed](https://nektosact.com/usage/index.html#github_token) when we run the workflow using nektos/act. The action for prettier requires authentication so we have pass it explicitly when running act. This can easily be done if you have the github cli installed like so:

```bash
act -s GITHUB_TOKEN="$(gh auth token)" pull_request
```
Or you can just generate a personal token and pass that.

The error with xvfb is because
>"The default images on nektos/act do not contain all the tools that GitHub Actions offers by default in their runners. Many things can work improperly or not at all while running those image." 

There are certain packages that electron needs to run tests that nektos default runner does not supply. So in order to be able to run our workflow you'll have to use a different docker environment. There are some [recommendations](https://nektosact.com/usage/runners.html#alternative-runner-images) outlined in the documentation for alternatives. Unfortunately those images require 18+ GBs of space which my computer does not have lol. Feel free to try it and see if it works @oestrada1001

Reply:
Great! Thank you for the very detailed update. 

I was able to run `act -s GITHUB_TOKEN="$(gh auth token)" pull_request` and I was able to get prettier to pass. I'm going to checkout the docker recommendation and give it a try.

I'll keep you updated. Great job @tolobayo

Reply:
Quick update: No Luck. I wasn't able to get Jest to pass using the suggested runners.

This is the command I used:
`act -s GITHUB_TOKEN="$(gh auth token)" -P ubuntu-18.04=catthehacker/ubuntu:full-18.04 pull_request`

I've attached the error log to this message.
[error.log](https://github.com/user-attachments/files/16152585/error.log)

I'll approve it and we can debug it while we merge it.